packages/core/components/DataTable/DataTable.tsx

```diff
@@ -89,8 +89,11 @@ const DataTable = (props: DataTableProps) => {
 
   const [expanded, setExpanded] = useState(expandDataTable)
 
+  const hasDateAxis = config.xAxis && ['date-time', 'date'].includes(config.xAxis.type)
```

Comment:
config. should be datatable config and therefore it doesn't know about any xAxis.